openmmtools/storage/iodrivers.py

```diff
@@ -0,0 +1,1653 @@
+            raise ValueError("Input data must have units of {}, but instead is {}".format(self._unit,
+                                                                                          str(data.unit)))
+        # Determine current current length and therefore the last index
+        length = self._bound_target.shape[0]
+        # Save data
+        # Strip Unit
+        data_unit = data.unit
+        data_value = data / data_unit
+        packaged_data = self._encoder(data_value)
+        self._bound_target[length, :] = packaged_data
+
+    def _determine_data_information(self, data):
+        # Make common _bind functions a single function
+        data_unit = data.unit
+        data_value = data / data_unit
+        data_type_name = typename(type(data_value))
+        try:
+            data_shape = data_value.shape
+            data_base_type = type(data_value.flatten()[0])
+            self._compare_shape = lambda x: x.shape
+        except AttributeError:  # Trap not array
+            try:
+                data_shape = (len(data_value),)
+                data_base_type = type(data_value[0])
+                self._compare_shape = lambda x: (len(x),)
+            except TypeError:  # Trap not iterable
+                data_shape = 1
+                data_base_type = type(data_value)
+                self._compare_shape = lambda x: 1
+        return data_shape, data_base_type, data_type_name
+
+    def _set_codifiers(self, stype):
+        # Assign the codecs in a single block
+        if stype == 'int':
+            self._encoder = simple_encoder
+            self._decoder = scalar_decoder_generator(int)
+        elif stype == 'float':
+            self._encoder = simple_encoder
+            self._decoder = scalar_decoder_generator(float)
+        elif stype == 'list' or stype == 'tuple':
+            self._encoder = simple_encoder
+            self._decoder = nc_iterable_decoder
+        elif 'ndarray' in stype:
+            self._encoder = simple_encoder
+            self._decoder = nc_numpy_array_decoder
+        else:
+            raise TypeError("NCQuantity does not know how to handle a quantity of type {}!".format(stype))
+
+    @property
+    def storage_type(self):
+        return 'variables'
+
+
+# =============================================================================
+# NETCDF DICT YAML HANDLERS
+# =============================================================================
+
+class NCYamlLoader(yaml.Loader):
+    """PyYAML Loader that recognized !Quantity nodes, converts YAML output -> Python type"""
+    def __init__(self, *args, **kwargs):
+        super(NCYamlLoader, self).__init__(*args, **kwargs)
+        self.add_constructor(u'!Quantity', self.quantity_constructor)
+
+    @staticmethod
+    def quantity_constructor(loader, node):
+        loaded_mapping = loader.construct_mapping(node)
+        data_unit = quantity_from_string(loaded_mapping['NCUnit'])
+        data_value = loaded_mapping['NCValue']
```

Comment:
No no this is fine for now, but if in the future we want to do this somewhere else, we'll need a dict to str serialization that has not `NC` in the name and we'll have to duplicate the code.